posthog/api/test/test_feature_flag.py

```diff
@@ -352,6 +453,138 @@ def test_deleting_feature_flag(self):
+        #         "rollout_percentage": None,
+        #     },
+        # )
+        # self.assertEqual(historical_flags[0].action, "delete")
+
+    @freeze_time("2021-08-25T22:09:14.252Z")
+    def test_get_feature_flag_that_needs_importing(self):
+        """
+        The first time we load history for a feature flag that existed before starting history logging
+        Import its current state
+        """
+        new_user = User.objects.create_and_join(self.organization, "new_annotations@posthog.com", None)
+
+        instance = FeatureFlag.objects.create(team=self.team, created_by=self.user)
+        self.client.force_login(new_user)
+
+        response = self.client.get(f"/api/projects/{self.team.id}/feature_flags/{instance.pk}/history")
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        history = response.json()["results"]
+
+        expected = [
+            asdict(
+                HistoryListItem(
+                    email="history.hog@posthog.com",
+                    name="history hog",
+                    action="FeatureFlag_imported",
+                    detail={"id": str(instance.pk), "key": ""},
+                    created_at="2021-08-25T22:09:14.252000+00:00",
+                )
+            )
+        ]
+
+        self.assertEqual(
+            history, expected,
+        )
+
+    def test_get_feature_flag_history(self):
+        """
+        The first time we load history for a feature flag that existed before starting history logging
+        Import its current state
+        """
+        new_user = User.objects.create_and_join(
+            self.organization, "person_acting_and_then_viewing_history@posthog.com", None
+        )
+        self.client.force_login(new_user)
+
+        with freeze_time("2021-08-25T22:09:14.252Z") as frozen_datetime:
+            create_response = self.client.post(
+                f"/api/projects/{self.team.id}/feature_flags/",
+                {"name": "feature flag with history", "key": "feature_with_history"},
+            )
+
+            self.assertEqual(create_response.status_code, status.HTTP_201_CREATED)
+            flag_id = create_response.json()["id"]
+
+            frozen_datetime.tick(delta=datetime.timedelta(minutes=10))
+
+            update_response = self.client.patch(
+                f"/api/projects/{self.team.id}/feature_flags/{flag_id}",
+                {
+                    "name": "feature flag with history",
+                    "filters": {"groups": [{"properties": [], "rollout_percentage": 74}]},
+                },
+                format="json",
+            )
+
+        self.assertEqual(update_response.status_code, status.HTTP_200_OK)
+
+        self.assert_feature_flag_history(
+            flag_id,
+            [
+                {
+                    "email": "person_acting_and_then_viewing_history@posthog.com",
+                    "name": "",
+                    "action": "FeatureFlag_filters_changed",
```

Comment:
Can we use something like `"action": "change" | "create" | "delete"` instead which we can use to determine the type of detail e.g. with union types?

We can pull out the key from the detail anyhow, and for history, at least for this change we know that this is FeatureFlag as we're fetching from feature_flags history endpoint

Reply:
hmmmm 🤔 

I'm anticipating the humanizing ending up as Map/Dict lookups

We could split action into a structure `{ type: FeatureFlag, key: filters, action: changed}`

Thinking about [this comment](https://github.com/PostHog/posthog/pull/8931#discussion_r826738883) we might end up with

`{ changes:  [{ type: FeatureFlag, key: filters, action: changed, detail: {stuff: here}}]}`

each item is distinct so could be typed or used as a key in a lookup